### ISSUE TITLE
Build 5.2.0.4 Source Question(s)

### ISSUE THREAD

**Issue by Dcreeron** [dynarithmic/twain_library_source#1]
Hey Paul,

**TLDR;** latest source code pull is missing some files in the DTWAIN_DLL project. They are on disk, but not in the .vcxproj. Skip to "Resolved" below if you want.

--

I grabbed the latest code via the git link and am trying to build it (VS2019 32-bit). I was going to cleanup the VS2019 files (realized there were some issues after my last attempt) and then take a shot at adding DTWAIN_* calls for the AutomaticSenseMedium capability. [I have this working through the DTWAIN_SetCap() calls but would prefer it 'built in'.]

When I try and build, after adding new vs2019 projects and fixing the build configurations/etc., I'm getting errors on the following methods in the dtwain32_d.def file:

DTWAIN_SetErrorCallback                           @30060
DTWAIN_SetErrorCallback64                         @30061
DTWAIN_GetErrorCallback                           @30062
DTWAIN_GetErrorCallback64                         @30063

I can't find any source for these, and they don't show up in the online documentation link.

Are these used/needed any more?

If I remove them from dtwainx.h and the dtwain32_d.def  I'm able to successfully build debug (no new DTWAIN_* methods yet).

NOTE: I'm currently using the binaries from github for 5.2.0.4 in my project, and I can call DTWAIN_SetErrorCallback(). Looks like it gets hit, but I'm not 100% sure why without some documentation/code.

Am I missing something?

**Resolved:**

Digging further.... it looks like ctrlerrorcallback.cpp isn't part of the project's source files (it's on disk, just not in the project). Neither is ctlskew.cpp.

NOTE: When I created my "vs2019" versions of the projects I copied the 2017 ones as my starting point.

When I add these files into the project, and put the methods back in the .def/dtwainx.h I'm able to build debug and release builds successfully. Tomorrow I'll try the *U builds and then actually use the binaries in my project.

After that I'll take a shot at adding the AutomaticSenseMedium calls.
